src/pages/AuthCallbackHandler.js

```diff
@@ -1,111 +1,142 @@
-                                // TODO: Check if account ID matches the one from email
-                                if (!accountCreationData.privateKey || !accId) throw ('Could not find account creation data');
-
-                                window.localStorage.setItem('fast-auth:account-creation-data', JSON.stringify({
-                                    ...accountCreationData,
-                                    accountId: accId,
-                                    isCreated: true
-                                }));
-
-                                setStatusMessage('Redirecting to app...');
-
-                                window.location.href = '/';
-                            });
-                    }
+  const history = useHistory();
+  const [statusMessage, setStatusMessage] = React.useState("Loading...");
+
+  React.useEffect(() => {
+    const locationUrl = window.location.href;
+
+    if (isSignInWithEmailLink(firebaseAuth, locationUrl)) {
+      const url = new URL(locationUrl);
+      const searchParams = new URLSearchParams(url.search);
+      const accountId = searchParams.get("accountId");
+      const publicKey = searchParams.get("publicKey");
+      const isRecovery = searchParams.get("isRecovery") === "true";
+
+      let email = window.localStorage.getItem("emailForSignIn");
+
+      setStatusMessage("Verifying email...");
+      signInWithEmailLink(firebaseAuth, email, window.location.href)
+        .then(async (result) => {
+          window.localStorage.removeItem("emailForSignIn");
+
+          const limitedAccessKey = nearAPI.KeyPair.fromRandom("ED25519");
+
+          const user = result.user;
+          if (!!user.emailVerified) {
+            setStatusMessage(
+              isRecovery ? "Recovering account..." : "Creating account..."
+            );
+
+            // TODO: Call MPC Service with accountId, publicKey,  and oauthToken to create account
+            const data = {
+              ...(accountId && accountId.includes(".")
+                ? { near_account_id: accountId }
+                : {}),
+              create_account_options: {
+                full_access_keys: [publicKey],
+                limited_access_keys: [
+                  {
+                    public_key: limitedAccessKey.getPublicKey().toString(),
+                    receiver_id: signInContractId,
+                    allowance: "250000000000000",
+                    method_names: "",
+                  },
+                ],
+              },
+              oidc_token: user.accessToken,
+            };
+
+            const headers = new Headers();
+            headers.append("Content-Type", "application/json");
+
+            const options = {
+              method: "POST",
+              mode: "cors",
+              body: JSON.stringify(data),
+              headers,
+            };
+
+            await fetch(
+              `${MPC_RECOVERY_URL}/${isRecovery ? "add_key" : "new_account"}`,
```

Comment:
I don't think `add_key` supports `CreateAccountOptions`. Do you need us to add it?